src/analysis/interpolation/qgsinterpolator.h

```diff
@@ -22,39 +22,76 @@
 #include "qgis_sip.h"
 #include "qgis_analysis.h"
 
-class QgsVectorLayer;
+class QgsFeatureSource;
 class QgsGeometry;
+class QgsFeedback;
 
-struct ANALYSIS_EXPORT vertexData
+/**
+ * Interpolation data for an individual source vertex.
+ * \since QGIS 3.0
+ */
+struct ANALYSIS_EXPORT QgsInterpolatorVertexData
```

Comment:
Do we have no class in core for this?

Reply:
Well... we could use QgsPoint. But I was concerned about the weight of it (extra double, vtable). I'll make this non-public API though.